overhead_analysis/run_tests.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+# Run fio tests on different modes: 0 - blockhisto disabled; 1 - blockhisto enabled
+# Place the fio job files in directory "job_files". 
+
+BCC=../bcc      # specify path to bcc script here
+
+
+function do_test {
+    mkdir -p $outdir 
+    for jobfile in ./job_files/*; do      # jobfile=./job_files/xxx.fio 
+        for job in $(grep -o "\[.*\]" $jobfile | awk -F "[\[\]]" '{print $2}'); do
+            if [ $job != "global" ]; then
+                echo "Run job: $job at file: $jobfile"
+                # stash cpu info
+                SECONDS=0
+                cpu_before=($(head -n1 /proc/stat | awk '{print $2,$3,$4,$5,$6,$7,$8,$9,$10,$11}'))
+
+                sudo fio $jobfile --section=$job --output=$outdir/${job}.out --output-format="json"
+
+                cpu_after=($(head -n1 /proc/stat | awk '{print $2,$3,$4,$5,$6,$7,$8,$9,$10,$11}'))
+                duration=$SECONDS
+                len=${#cpu_after[@]}
+                for ((i=0; i<$len; i++)); do
+                    cpu_after[i]=$(echo "scale=4; (${cpu_after[i]}-${cpu_before[i]})/$duration" |bc)
+                done
+                echo ${cpu_after[@]} > $outdir/${job}.perf
+            fi
+        done
+    done
+}
+
+
+if [ -z $1 ]
+then 
+    echo "Please input mode: 0 - blockhisto disabled; 1 - blockhisto enabled" 
+    exit 1
+fi 
+
+for cnt in 1 2
+do 
+    if [ $cnt -eq 1 -a $1 -eq 0 ]
+    then 
+        outdir=./output
+        echo "Run without tracing"
+
+    elif [ $cnt -eq 1 -a $1 -eq 1 ] 
+    then
+        outdir=./output-blkhist
```

Comment:
this file would probably be more generic if you specified outdir depending on if u were running blkhist kernel or not, instead of having extra arg just for that

Reply:
Indeed. Since for different tests there are different requirements for the script, this file is just for the test of blkhist/ebpf/noop.